sdk/servicebus/service-bus/test/public/sendAndSchedule.spec.ts

```diff
@@ -61,11 +61,19 @@ describe("Sender Tests", () => {
       : TestMessage.getSample();
     await sender.sendMessages(testMessage);
     const msgs = await receiver.receiveMessages(1);
+    // remove message first in case any assertion fails to ensure we don't have lingering message
+    await receiver.completeMessage(msgs[0]);
 
     should.equal(Array.isArray(msgs), true, "`ReceivedMessages` is not an array");
     should.equal(msgs.length, 1, "Unexpected number of messages");
     should.equal(msgs[0].deliveryCount, 0, "DeliveryCount is different than expected");
-    should.equal(msgs[0].state, "active");
+    should.equal(msgs[0].messageId, testMessage.messageId);
+    // the message could be in "scheduled" state because we set `scheduledEnqueueTimeUtc` property
+    should.equal(
+      msgs[0].state === "active" || msgs[0].state === "scheduled",
```

Comment:
Why not use the presence of `scheduledEnqueueTimeUtc` on the testMessage to determine the state on the recevied message?

Reply:
The received message might be in "active" or "scheduled" state, when a message is sent via `sendMessages()` with this property set.  Messages scheduled using `scheduleMessages()` will always be in "scheduled" state.

Reply:
Updated comments to clarify that it could be in either state.

Reply:
My concern is that if we received a message AFTER its scheduled time, then it should always be "active". The state is "scheduled" if we peek at it before its scheduled time. Is that not the right assumption?

Reply:
You are right that the message should have already transitioned to the "active" state, since we set the scheduled enqueue time to be current time when creating the message.  I suspect that the test is failing probably due to clock skew (so far only seeing this failure on macOS).  Since I already add a test for "active" state where we don't set the scheduled enqueue, maybe it's better to not verify state here?

Reply:
Yes, I would be fine not testing the state here